File: 26099/frontend/capture.py
import os, re
from playwright.sync_api import sync_playwright

PORT = 5174
BASE_URL = f"http://localhost:{PORT}"
REVIEW_DIR = r"C:\Users\udayp\Documents\code\SIH26\26099\frontend\review"

PAGES = [
    ("/", "Dashboard"),
    ("/overview", "Overview"),
    ("/explorer", "Explorer"),
    ("/matching", "Matching"),
    ("/duplicates", "Duplicates"),
    ("/savings", "Savings"),
    ("/registry", "Registry"),
    ("/migration", "Migration"),
    ("/integration", "Integration"),
    ("/import", "Import"),
    ("/normalize", "Normalize"),
    ("/activity", "Activity"),
    ("/engine", "Engine"),
]

os.makedirs(REVIEW_DIR, exist_ok=True)

with sync_playwright() as p:
    browser = p.chromium.launch(headless=True)
    page = browser.new_page(viewport={"width": 1440, "height": 900})
    page.goto(BASE_URL, wait_until="networkidle", timeout=30000)
    page.wait_for_timeout

    for path, label in PAGES:
        try:
            print(f"\n=== {label} ({path}) ===")
            page.goto(f"{BASE_URL}{path}", wait_until="networkidle", timeout=30000)
            page.wait_for_timeout

            safe = path.replace("/", "_").strip("_") or "home"
            out = os.path.join(REVIEW_DIR, f"{safe}_1.png")
            page.screenshot(path=out, full_page=False)
            print(f"Screenshot saved: {out}")

            body_text = page.inner_text("body")
            print(f"Page text preview: {body_text[:400]}")
        except Exception as e:
            print(f"FAILED {label}: {e}")

    browser.close()

print("\nDone!")

File: 26099/frontend/capture.mjs
import { chromium } from 'playwright';
import path from 'path';
import fs from 'fs';
import { fileURLToPath } from 'url';

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const BASE = 'http://localhost:59331';
const OUT = 'C:/Users/udayp/Documents/code/SIH26/26099/frontend/review';
if (!fs.existsSync(OUT)) fs.mkdirSync(OUT, { recursive: true });

(async () => {
  const browser = await chromium.launch();
  const page = await browser.newPage({ viewport: { width: 1400, height: 900 } });

  // Login
  await page.goto(BASE + '/login', { waitUntil: 'domcontentloaded' });
  await page.waitForTimeout(600);
  await page.getByPlaceholder('Your name (optional)').fill('Admin');
  await page.waitForTimeout(200);
  await page.getByRole('button', { name: 'Admin' }).click();
  await page.waitForTimeout(200);
  await page.getByRole('button', { name: 'Sign in' }).click();
  await page.waitForTimeout(800);

  // Capture login and dashboard
  await page.screenshot({ path: path.join(OUT, '01-login.png') });
  await page.screenshot({ path: path.join(OUT, '02-dashboard.png') });

  // Navigate to each page via URL
  const routes = [
    { url: '/overview', file: '03-overview.png' },
    { url: '/explorer', file: '04-explorer.png' },
    { url: '/matching', file: '05-matching.png' },
    { url: '/duplicates', file: '06-duplicates.png' },
    { url: '/savings', file: '07-savings.png' },
    { url: '/registry', file: '08-registry.png' },
    { url: '/migration', file: '09-migration.png' },
    { url: '/integration', file: '10-integration.png' },
    { url: '/import', file: '11-import.png' },
    { url: '/normalize', file: '12-normalize.png' },
    { url: '/activity', file: '13-activity.png' },
    { url: '/engine', file: '14-engine.png' },
  ];

  for (const r of routes) {
    await page.goto(BASE + r.url, { waitUntil: 'domcontentloaded' });
    await page.waitForTimeout(600);
    await page.screenshot({ path: path.join(OUT, r.file) });
    console.log('Captured:', r.file);
  }

  await browser.close();
  console.log('Done. All screenshots in', OUT);
})();
